megalinter/Linter.py

```diff
@@ -414,7 +410,16 @@ def load_reporters(self):
     # Collect all files that will be analyzed by the current linter
     def collect_files(self, all_files):
         # Filter all files to keep only the ones matching with the current linter
+
+        file_extensions = set(self.file_extensions)
+        file_names_regex = f'^{"$|^".join(self.file_names)}$'
```

Comment:
Please not here I emulate "full match" for regular expressions. Prepend by `^` and append `$`.
So "Dockerfile" will match only "Dockerfile".

Reply:
about Dockerfile, why not directly set these regexes in the dockerfile.yml descriptor ?

Reply:
1. I understood the problem in my code (I broke checking file content and some other logic).
Now I have fixed.

2. My implementation allows specifying regular expression inside of dockerfile.yml descriptor.
I know 2 ways of applying regular expression to a string:
- search for regex matching substring
- check if the whole string matches regex. I.e. the whole text should be matched.

I chose the second way because:
- many users will forget to add `^` and `$`.
- adding `^` and `$` twice does not change logic
- no need to modify anything if you need to match by full file name like `Dockerfile`
- it still allows partial match by adding `.*` around the regex

Reply:
I agree with your choice of option 2 :)